### DISCUSSION TITLE
App router body size limit per route

### DISCUSSION THREAD

**Discussion by Cajuteq** [vercel/next.js#68409] [App Router]
### Goals

1. Better documentation for body size limit
2. Switch from PageRouter to AppRouter
3. Allow non global override

### Non-Goals

1.
2.
3.

### Background

Two issues were already raised on this topic, but both went stale so I think a discussion might suits the topic better. See
https://github.com/vercel/next.js/issues/52457
https://github.com/vercel/next.js/issues/57501

In page router there was a configuration per page looking like :
```
export const config = {
  api: {
    bodyParser: {
      sizeLimit: "5mb",
    },
  },
};
```
https://nextjs.org/docs/pages/building-your-application/routing/api-routes#custom-config

In App router, since https://github.com/vercel/next.js/discussions/49891 there is a global `experimental.serverActions.bodySizeLimit`
https://nextjs.org/docs/app/api-reference/next-config-js/serverActions

During convertion to App Router, keeping the old option triggers an error with a link to https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config which list the current options. One can recognize maxDuration which is present in both the old and the new config.

### Proposal

I think the linked page should at least provide a link or a hint to the new global (experimental) option (which we found difficult to find)

We only used to override the sizeLimit on specific route who are dealing with files. We no longer have this possibility are now constrained to allow it globaly. I think we should keep the new global parameter but also reintroduce the per Route override option similar to how maxDuration now has both.

The documentation could then be updated in a way similar to maxDuration
https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#maxduration
which provide a "Good to know" section with a link to the global parameter.

**Comment by tremby** [comment]
Related to this, I'd like also like to be able to set custom limits per *server action*.
